File: tests/fixtures/codegen/expected/typescript/runtime/queries/events/ListEvents.ts
import type { ListEventsParams, ListEventsRow } from "../../../types/queries/events/ListEvents.js";
export type { ListEventsParams, ListEventsRow } from "../../../types/queries/events/ListEvents.js";
import * as pgsid from "./pgsid/queryable.js";
export { QueryValidationError } from "./pgsid/queryable.js";
import * as _jsonSchemas from "../../jsonschemas/index.js";
export const listEventsSql = "WITH selected AS (SELECT id, payload, note FROM public.events)\nSELECT id, payload, note FROM selected ORDER BY id;";
export function isListEventsPayload(value: unknown): value is ListEventsRow["payload"] {
    return _jsonSchemas.isEventPayload(value);
}
export function validateListEventsPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEventPayload(value);
}
export async function listEvents(db: pgsid.Queryable): Promise<ListEventsRow[]> {
    const result = await db.query(listEventsSql, []);
    return result.rows.map(row => {
        if (!isListEventsPayload(row["payload"]))
            throw new pgsid.QueryValidationError("ListEvents", "payload", validateListEventsPayload(row["payload"]).issues);
        return row as ListEventsRow;
    });
}

File: tests/fixtures/codegen/expected/go/queries/events/listevents.go
package events

import (
	"context"
	jsonschemas "example.com/pgsid-fixture/generated/go/jsonschemas"
	public "example.com/pgsid-fixture/generated/go/schema/public"
)

const ListEventsSQL = "WITH selected AS (SELECT id, payload, note FROM public.events)\nSELECT id, payload, note FROM selected ORDER BY id;"

type ListEventsParams struct {
}
type ListEventsRow struct {
	Id      public.EventId           `db:"id"`
	Payload jsonschemas.EventPayload `db:"payload"`
	Note    *string                  `db:"note"`
}

func (q *Queries) ListEvents(ctx context.Context) ([]ListEventsRow, error) {
	rows, err := q.db.Query(ctx, ListEventsSQL)
	if err != nil {
		return nil, err
	}
	defer rows.Close()
	items := make([]ListEventsRow, 0)
	for rows.Next() {
		var row ListEventsRow
		if err := rows.Scan(&row.Id, &row.Payload, &row.Note); err != nil {
			return nil, err
		}
		items = append(items, row)
	}
	if err := rows.Err(); err != nil {
		return nil, err
	}
	return items, nil
}
